keras_cv/models/backbones/mobilenet_v3/mobilenet_v3_backbone.py

```diff
@@ -0,0 +1,446 @@
+# Copyright 2023 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""MobileNet v3 backbone model.
+
+References:
+    - [Searching for MobileNetV3](https://arxiv.org/pdf/1905.02244.pdf)
+    (ICCV 2019)
+    - [Based on the original keras.applications MobileNetv3](https://github.com/keras-team/keras/blob/master/keras/applications/mobilenet_v3.py)
+"""  # noqa: E501
+
+import copy
+
+from tensorflow import keras
+from tensorflow.keras import backend
+from tensorflow.keras import layers
+from tensorflow.keras.utils import custom_object_scope
+
+from keras_cv import layers as cv_layers
+from keras_cv.models import utils
+from keras_cv.models.backbones.backbone import Backbone
+from keras_cv.models.backbones.mobilenet_v3.mobilenet_v3_backbone_presets import (  # noqa: E501
+    backbone_presets,
+)
+from keras_cv.utils.python_utils import classproperty
+
+CHANNEL_AXIS = -1
+BN_EPSILON = 1e-3
+BN_MOMENTUM = 0.999
+
+
+def adjust_channels(x, divisor=8, min_value=None):
+    """Ensure that all layers have a channel number divisible by the `divisor`.
+
+    Args:
+        x: integer, input value.
+        divisor: integer, the value by which a channel number should be
+            divisible, defaults to 8.
+        min_value: float, optional minimum value for the new tensor. If None,
+            defaults to value of divisor.
+
+    Returns:
+        the updated input scalar.
+    """
+
+    if min_value is None:
+        min_value = divisor
+
+    new_x = max(min_value, int(x + divisor / 2) // divisor * divisor)
+
+    # make sure that round down does not go down by more than 10%.
+    if new_x < 0.9 * x:
+        new_x += divisor
+    return new_x
+
+
+def apply_hard_sigmoid(x):
+    """
+    Args:
+        x: input tensor
+    Returns:
+        the updated input tensor.
+    """
+
+    activation = layers.ReLU(6.0)
+
+    return activation(x + 3.0) * (1.0 / 6.0)
+
+
+def apply_hard_swish(x):
+    """
+    Args:
+        x: input tensor
+
+    Returns:
+        the updated input tensor.
+    """
+
+    multiply_layer = layers.Multiply()
+
+    return multiply_layer([x, apply_hard_sigmoid(x)])
+
+
+def apply_inverted_res_block(
+    x,
+    expansion,
+    filters,
+    kernel_size,
+    stride,
+    se_ratio,
+    activation,
```

Comment:
We want to be able to pass string args for the activations so that our presets do not depend on the backbone code. We can already pass `"relu"` as a string, but could you add special handling for the `"hard_swish"` string?

```python
if activation == "hard_swish":
    activation = apply_hard_swish()
```